File: java2csharp/java/Samsung/Camera/util/SettingItem.java
package com.samsung.android.sdk.camera.sample.cases.util;

import android.app.Activity;
import android.view.View;

/**
 * Abstract Class for SettingItem
 */
public abstract class SettingItem<K, V> {
    public static final int SETTING_TYPE_REQUEST_KEY = -1;
    public static final int SETTING_TYPE_CAMERA_FACING = 0;
    public static final int SETTING_TYPE_IMAGE_FORMAT = 1;

    protected final Activity mActivity;
    protected final String mName;
    protected final K mKey;

    protected OnSettingItemValueChangedListener mListener;

    protected View mView;

    public SettingItem(Activity activity, String name, K key) {
        mActivity = activity;
        mName = name;
        mKey = key;
    }

    /**
     * Returns current setting value for this setting item.
     */
    public abstract V getValue();

    /**
     * Sets value for this setting item.
     */
    public abstract void setValue(V value);

    /**
     * Set initial setting value for this setting item.
     */
    @SuppressWarnings("unchecked")
    public void setInitialValue(Object value) {
        if(value != null) setValue((V)value);
        if(mListener != null) mListener.onSettingItemValueChanged(mKey, getValue());
    }

    /**
     * Returns key for this setting item.
     */
    public K getKey() { return mKey; }

    /**
     * Returns view for this setting item.
     */
    public View getView() { return mView; }

    /**
     * Enables/Disables this setting item.
     */
    public abstract void setEnabled(boolean enable);

    /**
     * Returns whether this setting item is enabled.
     */
    public abstract boolean isEnabled();

    /**
     * Set {@link SettingItem.OnSettingItemValueChangedListener} for this setting item.
     */
    public void setOnSettingItemValueChangedListener(OnSettingItemValueChangedListener listener) { mListener = listener; }

    public interface OnSettingItemValueChangedListener {
        public <K_Listener, V_Listener> void onSettingItemValueChanged(K_Listener key, V_Listener value);
    }
}

File: java2csharp/csharp/Samsung/Camera/cases/util/SettingItem.cs
namespace com.samsung.android.sdk.camera.sample.cases.util
{

	using Activity = android.app.Activity;
	using View = android.view.View;

	/// <summary>
	/// Abstract Class for SettingItem
	/// </summary>
	public abstract class SettingItem<K, V>
	{
		public const int SETTING_TYPE_REQUEST_KEY = -1;
		public const int SETTING_TYPE_CAMERA_FACING = 0;
		public const int SETTING_TYPE_IMAGE_FORMAT = 1;

		protected internal readonly Activity mActivity;
		protected internal readonly string mName;
		protected internal readonly K mKey;

		protected internal OnSettingItemValueChangedListener mListener;

		protected internal View mView;

		public SettingItem(Activity activity, string name, K key)
		{
			mActivity = activity;
			mName = name;
			mKey = key;
		}

		/// <summary>
		/// Returns current setting value for this setting item.
		/// </summary>
		public abstract V Value {get;set;}


		/// <summary>
		/// Set initial setting value for this setting item.
		/// </summary>
//JAVA TO C# CONVERTER TODO TASK: Most Java annotations will not have direct .NET equivalent attributes:
//ORIGINAL LINE: @SuppressWarnings("unchecked") public void setInitialValue(Object value)
		public virtual object InitialValue
		{
			set
			{
				if (value != null)
				{
					Value = (V)value;
				}
				if (mListener != null)
				{
					mListener.onSettingItemValueChanged(mKey, Value);
				}
			}
		}

		/// <summary>
		/// Returns key for this setting item.
		/// </summary>
		public virtual K Key
		{
			get
			{
				return mKey;
			}
		}

		/// <summary>
		/// Returns view for this setting item.
		/// </summary>
		public virtual View View
		{
			get
			{
				return mView;
			}
		}

		/// <summary>
		/// Enables/Disables this setting item.
		/// </summary>
		public abstract bool Enabled {set;get;}


		/// <summary>
		/// Set <seealso cref="SettingItem.OnSettingItemValueChangedListener"/> for this setting item.
		/// </summary>
		public virtual void setOnSettingItemValueChangedListener(OnSettingItemValueChangedListener listener)
		{
			mListener = listener;
		}

		public interface OnSettingItemValueChangedListener
		{
			void onSettingItemValueChanged<K_Listener, V_Listener>(K_Listener key, V_Listener value);
		}
	}

}